tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_device_info.py

```diff
@@ -13,10 +13,16 @@ def test_that_device_info_is_populated(self):
         settings.launch()
         device_info = settings.open_device_info_settings()
 
+        # devices without sim cards have no phone number
+        if 'local_phone_numbers' in self.testvars:
+            self.assertTrue(len(device_info.phone_number) > 0)
+            self.assertIsNotNone(device_info.phone_number)
+        else:
+            self.assertEqual(device_info.phone_number, '')
```

Comment:
A concern here is the template for our testvars has a "local_phone_numbers": [""] as the default, so would foul this if someone assumed that was OK to leave. 

  if ('local_phone_numbers' in self.testvars and 
      bool(self.testvars.local_phone_numbers) and 
      self.testvars.local_phone_numbers[0] != ""):

...is probably about as robust as I can think of. Also now verbose enough that it should go in a module somewhere. :)

On another note, realized my own local_phone_numbers check will error if a number isn't defined, so I have to update the ftu PR similarly.

Also, is there a situation where len(x) > 0 and x is None? If not (as len(None) throws an exception) you can remove line 19.